socs/agents/starcam_lat/agent.py

```diff
@@ -0,0 +1,262 @@
+        self.agent.register_feed("starcamera", record=True,
+                                 agg_params=agg_params, buffer_time=1)
+        try:
+            self.StarcamHelper = StarcamHelper(ip_address, port)
+        except socket.timeout:
+            self.log.error("Starcamera connection has times out")
+            return False, "Timeout"
+
+    @ocs_agent.param('_')
+    def send_commands(self, session, params=None):
+        """
+        send_commands()
+
+        **Process**
+        packs and sends camera+astrometry-related commands to starcam
+
+        **Return**
+        returns a touple with True/False and a string describing whether
+        or not a lock could be acquired and commands were sent to the sc
+        """
+        with self.lock.acquire_timeout(job='send_commands') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self._lock.job} is already running")
+                return False, "Could not acquire lock"
+            self.log.info("Sending commands")
+            self.StarcamHelper.pack_and_send_cmds()
+        return True, "Sent commands to starcamera"
+
+    @ocs_agent.param('_')
+    def acq(self, session, params=None):
+        """
+        acq()
+
+        **Process**
+        acquires data from starcam and publishes to feed
+
+        **Return**
+        once the acq() loop exits (wherein data is retrieved from
+        the camera and pulished), a touple with True/False and a string
+        describing whether or not the loop was exited after the end of
+        an acquisition.
+        """
+        if params is None:
+            params = {}
+        with self.lock.acquire_timeout(timeout=100, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because {} is already "
+                              "running".format(self.lock.job))
+                return False, "Could not acquire lock"
+            session.set_status('running')
+            self.log.info("Starting acquisition")
+            self.take_data = True
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'astrometry',
+                    'data': {}
+                }
+                # get astrometry data
+                astrom_data_dict = self.StarcamHelper.get_astrom_data()
+                # update the data dictionary+session and publish
+                data['data'].update(astrom_data_dict)
+                session.data.update(data['data'])
+                self.agent.publish_to_feed('starcamera', data)
+
+        return True, 'Acquisition exited cleanly'
+
+    def _stop_acq(self, session, params):
+        ok = False
+        if self.take_data:
+            session.set_status('stopping')
+            self.take_data = False
+            ok = True
+            # self.StarcamHelper.close()
```

Comment:
Can remove this comment. We don't want to close the TCP connection when stopping `acq`, since as written there's no way to re-establish that connection without restarting the agent.